source/utilities.cc

```diff
@@ -2277,6 +2277,174 @@ namespace aspect
 
 
 
+    template <int dim>
+    AsciiDataLayered<dim>::AsciiDataLayered ()
+    {}
+
+
+
+    template <int dim>
+    void
+    AsciiDataLayered<dim>::initialize(const unsigned int components)
+    {
+      AssertThrow ((dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()))
+                   || (dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model())) != nullptr
+                   || (dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model())) != nullptr
+                   || (dynamic_cast<const GeometryModel::Box<dim>*> (&this->get_geometry_model())) != nullptr,
```

Comment:
I know this is still common in other parts of the code, but could you please replace this with the new and easier structure: `Plugins::plugin_type_matches<GeometryModel::Box<dim> >(this->get_geometry_model())` ? Same for all the other checks.